src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,793 @@
+    this.#maxInlineSize = fromPxToNumber(computedStyle.maxInlineSize);
+    this.#minBlockSize = fromPxToNumber(computedStyle.minBlockSize);
+    this.#minInlineSize = fromPxToNumber(computedStyle.minInlineSize);
+
+    // Avoid watching border changes during the resize
+    this.#removeBorderSizeWatcher();
+
+    // Avoid listener on document click
+    document.removeEventListener("click", this.#evaluateClickOnDocument, {
+      capture: true
+    });
+
+    // Add listeners
+    document.addEventListener("mousemove", this.#trackElementResizeRAF, {
+      capture: true
+    });
+
+    document.addEventListener("mouseup", this.#handleResizeEnd, {
+      capture: true,
+      passive: true
+    });
+  };
+
+  #trackElementResizeRAF = (event: MouseEvent) => {
+    this.#resizeRAF.perform(this.#trackElementResize, () => {
+      // Improve drag UX by not selecting any button or clicking interactive
+      // elements
+      event.preventDefault();
+
+      // We remove the pointer-events and user-select properties after the first
+      // "mousemove", otherwise double clicking to select text would not work
+      this.#addDraggingClass();
+
+      this.#lastDragEvent = event;
+    });
+  };
+
+  #trackElementResize = () => {
+    // - - - - - - - - - - - - - DOM read operations - - - - - - - - - - - - -
+    const dialogRect = this.#dialogRef.getBoundingClientRect();
+
+    // - - - - - - - - - - - - - DOM write operations - - - - - - - - - - - - -
+    this.#resizeEdgesAndCornersDictionary[this.#currentEdge](dialogRect);
+
+    // Update last point
+    this.#initialDragEvent = this.#lastDragEvent;
+  };
+
+  #handleResizeEnd = () => {
+    if (this.adjustPositionAfterResize) {
+      this.#unfixDialogPosition();
+    } else {
+      this.#fixDialogPosition();
+    }
+
+    this.resizing = false;
+
+    // Cancel RAF to prevent access to undefined references
+    if (this.#resizeRAF) {
+      this.#resizeRAF.cancel();
+    }
+
+    // Reset document cursor back to normal
+    document.body.style.cursor = null;
+
+    // Reset dragged distance to its original value
+    setProperty(this.el, DIALOG_DRAGGED_X, this.#draggedDistanceX);
+    setProperty(this.el, DIALOG_DRAGGED_Y, this.#draggedDistanceY);
+
+    // Start again watching border size changes
+    this.#setBorderSizeWatcher();
+
+    // Remove listeners
+    document.removeEventListener("mousemove", this.#trackElementResizeRAF, {
+      capture: true
+    });
+
+    document.removeEventListener("mouseup", this.#handleResizeEnd, {
+      capture: true
+    });
+
+    this.#removeDraggingClass();
+
+    // Free the memory
+    this.#resizeRAF = null;
+    this.#initialDragEvent = null;
+    this.#lastDragEvent = null;
+
+    // Avoid listener on document click
+    this.modal &&
+      requestAnimationFrame(() => {
+        document.addEventListener("click", this.#evaluateClickOnDocument, {
+          capture: true
+        });
+      });
```

Comment:
Don't use logical evaluation in this way. Is more clear to read:

```ts
if (this.modal) {
  ...
}
```

Bundlers optimize the code so we can write more readable code.

Also, add a description of why we need the `requestAnimationFrame` call.